app/models/minion.rb

```diff
@@ -18,6 +18,8 @@ def assign_role(role:)
       salt.assign_role role: role
     end
     true
+  rescue
```

Comment:
Since we know which exception we want to rescue from we'd better be explicit here:

```rescue Net::HTTPBadResponse```